### ISSUE TITLE
error: invalid memory reference when running cargo test --features embed-all --test utilities_tests

### ISSUE THREAD

**Issue by 3togo** [podo-os/dlib-face-recognition#13]
Below is the error message when I use rust v.1.67. No error when using rust v.1.62.

error: test failed, to rerun pass `--test utilities_tests`

Caused by:
  process didn't exit successfully: `/work/gitee/dlib-face-recognition/target/debug/deps/utilities_tests-efe1d9d451438d32` (signal: 11, SIGSEGV: invalid memory reference)

**Comment by HoKim98** [comment]
Thanks for raising the issue with a good diagnosis.

I have confirmed that the same problem occurs on my personal Linux PC. The cause needs to be found. When I have time, I'll diagnose what's wrong with the build process.

**Comment by JSanchesDovichi** [comment]
Hello!
Bug reproduced on Linux(x64 - Arch).

Took a quick look at this, and narrowed down the problem a bit. Hadn't had time to try anything to fix, but here is some important things:

It definitely is a problem with the CPP macro call in the code. Using rust-analyzer, the code breaks entirely whenever i change anything inside the macro call. Seeing as the entire source code uses these calls, it's safe to bet that everything else is broken too.

The bug in question only happens whenever some code that is inside these macros is executed. That's the reason why the invalid_array_size test works, as the from_vec function breaks off with invalid sizes before calling the macro.

Tried some quick possibilities as updating CPP crate dependecy, but didn't work.

Will keep investigating, but i hope this helps!

**Comment by JSanchesDovichi** [comment]
Little update:
Curiously, the example script runs fine on stable rust, so the link doesn't really fail(? - pending confirmation)
Seems like it is a problem with linking in the tests. Can confirm that both test files aren't working, but the tests inside the crates are working.

Edit: After some more digging, i can confirm that only the from_vec() function in encoding.rs file is throwing this error.  I wrote that a while ago using new_from_scalar() as base, however, something seems to be broken in it now, and it's not working. I also confirmed that new_from_scalar() is working, as well as the rest of the crate too, including the benchmark.rs tests.
